### PR TITLE
Clearing all tags

### PR DESCRIPTION
**Pull request by BartSoftware**

### CODE REVIEW

File: app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivityTest.java [hunk 1]
```diff
@@ -108,6 +108,18 @@ public void whenOptionItemIsSelectedAndTagsAreNotChanged_OnlyBackToImageActivity
     }
 
     @Test
+    public void whenClearAllTagsIsCalled_OnlyBackToImageActivityIsCalled() {
+        MenuItem menuItem = mock(MenuItem.class);
```
Comment (michal1990) [inline app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivityTest.java]: missing Given

File: app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivityTest.java [hunk 2]
```diff
@@ -108,6 +108,18 @@ public void whenOptionItemIsSelectedAndTagsAreNotChanged_OnlyBackToImageActivity
...
+        when(menuItem.getItemId()).thenReturn(2);
+
+        //when
```
Comment (michal1990) [inline app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivityTest.java]: //When

File: app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivityTest.java [hunk 3]
```diff
@@ -108,6 +108,18 @@ public void whenOptionItemIsSelectedAndTagsAreNotChanged_OnlyBackToImageActivity
...
+        activity.onOptionsItemSelected(menuItem);
+
+        //then
```
Comment (michal1990) [inline app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivityTest.java]: //Then

**Review by michal1990 — COMMENTED** [review]
Please don't merge this PR before PR with Rational types do. Rational type tags are not cleared.

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivity.java [hunk 4]
```diff
@@ -149,6 +152,9 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 saveTags();
                 startImagesActivity();
                 return true;
+            case CLEAR_ALL_TAGS:
+                clearAllTags();
+                return true;
```
Comment (kejmil01) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivity.java:159]: We should think about moving toolbar (buttons) logic to separate viewModel-like class. This is only a suggestion to think about.

File: app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivityTest.java [hunk 5]
```diff
@@ -108,6 +108,19 @@ public void whenOptionItemIsSelectedAndTagsAreNotChanged_OnlyBackToImageActivity
     }
 
     @Test
+    public void whenClearAllTagsIsCalled_OnlyBackToImageActivityIsCalled() {
+        //Given
+        MenuItem menuItem = mock(MenuItem.class);
+        when(menuItem.getItemId()).thenReturn(2);
```
Comment (kejmil01) [inline app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/EditActivityTest.java]: I think that using static field from EditActivity (visibility protected for test) would be better than just plain integer value.

**Review by kejmil01 — APPROVED** [review]

**Review by michal1990 — APPROVED** [review]

**Pull request merged**